packages/attestation-service/src/index.ts

```diff
@@ -40,12 +48,23 @@ async function init() {
   app.get('/healthz', rateLimiter, asyncHandler(handleLivenessRequest))
   app.post(
     '/attestations',
+    express.json(),
     createValidatedHandler(AttestationRequestType, handleAttestationRequest)
   )
   app.post(
     '/test_attestations',
+    express.json(),
     createValidatedHandler(AttestationServiceTestRequestType, handleTestAttestationRequest)
   )
+
+  for (const p of smsProvidersWithDeliveryStatus()) {
+    const path = `/delivery_status_${p.type}`
+    rootLogger.info(
+      { url: deliveryStatusURLForProviderType(p.type) },
+      'Registered delivery status handler'
+    )
+    app.post(path, ...p.deliveryStatusHandlers(), handleAttestationDeliveryStatus(p.type))
```

Comment:
The flow is the same for both. Twilio and Nexmo will asynchronously callback to us via a POST to this endpoint. Then the handler parses that response, and calls `receivedDeliveryReport` which looks for an entry with a matching `deliveryId` in the `ongoingDeliveries` map. If it's a failure, this may cause us to retry delivery with the same provider (up to the retry limit) or after that with the next provider, until all retries are exhausted.

Reply:
got it -- i guess the confusion is that with Twilio the handlers are `[bodyParser.urlencoded({ extended: false }), twilio.webhook()]` vs Nexmo `[bodyParser.json()]`.  

what exactly is happening with the Twilio handlers?  it suggests that Twilio will call this endpoint with an encoded url that the twilio.webhook() handler uses to retrieve the receipt.  is that a correct understanding?

Reply:
oh I see. `twilio.webhook()` is actually something totally separate - it allows you to auth the signature in the incoming requests

Reply:
what happens after the incoming request gets parsed as a url?